src/test/java/core/basesyntax/strategy/handlers/BalanceOperationHandlerTest.java

```diff
@@ -0,0 +1,52 @@
+package core.basesyntax.strategy.handlers;
+
+import core.basesyntax.db.dao.StorageDao;
+import core.basesyntax.db.dao.StorageDaoImp;
+import core.basesyntax.model.GoodsOperation;
+import java.util.HashMap;
+import java.util.Map;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+class BalanceOperationHandlerTest {
+    private static Map<String, Integer> storage;
+    private static StorageDao storageDao;
+    private static BalanceOperationHandler balanceOperationHandler;
+    private static GoodsOperation balanceOperation;
+
+    @BeforeAll
+    static void beforeAll() {
+        storage = new HashMap<>();
+        storageDao = new StorageDaoImp(storage);
+        balanceOperationHandler = new BalanceOperationHandler(storageDao);
+        balanceOperation = new GoodsOperation(GoodsOperation.TransactionType.BALANCE,
+                "banana", 50);
+    }
+
+    @AfterEach
+    void tearDown() {
+        storage.clear();
+    }
+
+    @Test
+    void handleOperation_validOperations_ok() {
+        balanceOperationHandler.handleOperation(balanceOperation);
+        Integer actual = storage.get(balanceOperation.getItem());
+        Integer expected = balanceOperation.getQuantity();
+        Assertions.assertEquals(expected, actual,
+                String.format("Expected quantity of %s not equals actual",
+                        balanceOperation.getItem()));
+    }
+
+    @Test
+    void handleOperation_nullItemName_notOk() {
+        balanceOperation = new GoodsOperation(balanceOperation.getTransactionType(),
+                null,
+                balanceOperation.getQuantity());
+        Assertions.assertThrows(RuntimeException.class,
```

Comment:
It's better to throw more specific exception or check exception message like example bellow:

@Test
public void whenExceptionThrown_thenAssertionSucceeds() {
    Exception exception = assertThrows(NumberFormatException.class, () -> {
        Integer.parseInt("1a");
    });

    String expectedMessage = "For input string";
    String actualMessage = exception.getMessage();

    assertTrue(actualMessage.contains(expectedMessage));
}

Reply:
Cool advice! Thank you!